MitoMathHelper.js

```diff
@@ -172,4 +172,62 @@ const MitoMathHelper = class MitoMathHelper {
         let normalizedDistance = aToPDoToAToB / aToBSquared;
         return [linePointA[0] + (aToB[0] * normalizedDistance), linePointA[1] + (aToB[1] * normalizedDistance)]
     }
+
+    /**
+     * Returns the dot product.
+     * @param vectorA
+     * @param vectorB
+     * @returns {number}
+     * @private
+     */
+    static _dotProduct(vectorA, vectorB) {
+        return vectorA[0] * vectorB[0] + vectorA[1] * vectorB[1];
+    }
+
+    /** Uses a math formula to find the resulting translational velocity of one vector after an elastic collision.
+     * @param massA
+     * @param positionA
+     * @param velocityA
+     * @param massB
+     * @param positionB
+     * @param velocityB
+     * @returns {[]}
+     * @private
+     */
+    static _getOneTranslationalVelocityAfterElasticCollision(massA, positionA, velocityA, massB, positionB, velocityB) {
+        //TODO: refactor variables names to whatever mr boss boy tells me to
+        let partOne = (2 * massB) / (massA + massB);
+
+        let velocityDifference = [velocityA[0] - velocityB[0], velocityA[1] - velocityB[1]];
+        let positionDifference = [positionA[0] - positionB[0], positionA[1], positionB[1]];
+        let velDiffPosDiffDotProduct = MitoMathHelper._dotProduct(velocityDifference, positionDifference);
+        let positionMagnitudeSquared =  positionDifference[0] * positionDifference[0] + positionDifference[1] * positionDifference[1];
+
+        let partTwo = partOne * velDiffPosDiffDotProduct / positionMagnitudeSquared;
+        let finalVector = [positionDifference[0] * partTwo, positionDifference[1] * partTwo];
+
+        return [velocityA[0] - finalVector[0], velocityA[1] - finalVector[1]];
+    }
+
+    /**
+     * Returns both resulting translational velocities after collision.
+     * @param bodyA
+     * @param bodyB
+     * @param collisionPoint
+     * @returns {[Array, Array]}
+     * @private
+     */
+    static _getTranslationalVelocitiesAfterElasticCollision(bodyA, bodyB, collisionPoint) {
```

Comment:
should be a public function, called like
```
static getMitoPhysicsBodyVelocitiesAfterCollision
```

to match how we do every other public function. also I dont think you need to specify elastic.